src/workloads/query/CsiFragmentedInsertsFlat.yml

```diff
@@ -0,0 +1,271 @@
+  SchemaWidth: &schemaWidth 10000
+  ObjectWidth: &objectWidth 10
+  SampleSize: &sampleSize 100  # want: sampleSize * objectWidth ~ 0.1 * schemaWidth
+
+ActorTemplates:
+- TemplateName: InsertFromSample
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "Insert"}}
+    Type: SamplingLoader
+    Threads: {^Parameter: {Name: "Threads", Default: 1}}
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Database: *db
+          Collection: *coll
+          # The 'SamplingLoader' actor re-samples on repeat, meaning that it would get a new set of
+          # documents likely with different paths. This makes each repeat hit different parts of the
+          # index, causing a long warm up tail.
+          Repeat: 1
+          # Sample size of 100 means that no more than 100*objectWidth paths will be "touched" by
+          # "the sample.
+          SampleSize: *sampleSize
+          # When the batch size is the same as sample size, each sampled document will be used in
+          # the batch exactly once, and this will be repeated for the number of batches. We expect,
+          # that the first batch would exhibit "cold" cache behaviour and the subsequent batches
+          # would be fully warmed up.
+          InsertBatchSize: *sampleSize
+          # The instances of the template should explicitly specify the number of batches to make it
+          # clear what they are testing wrt to cold/warm scenario. Using the default of 0 is UB.
+          # The stats per batch are available in 'IndividualBulkInsert' measurement.
+          Batches: {^Parameter: {Name: "Batches", Default: 0}}
+
+- TemplateName: SettleDown
+  Config:
+    # We are not interested in measuring the quiese phases, so it's OK to use the same name for all
+    # of them.
+    Name: Quiese
+    Type: QuiesceActor
+    Threads: 1
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: 1
+
+Actors:
+- Name: Setup
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *maxPhases
+      PhaseConfig:
+        Repeat: 1
+        Thread: 1
+        Database: admin
+        Operations:
+        - OperationName: AdminCommand
+          OperationCommand:
+            setParameter: 1
+            internalQueryFrameworkControl: "trySbeEngine"
+
+- Name: Loader
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *maxPhases
+      PhaseConfig:
+        # Cannot have more threads than the actor itself.
+        Threads: 1
+        Repeat: 1
+        Database: *db
+        CollectionCount: 1
+        DocumentCount: *docCount
+        BatchSize: 1000
+        Document:
```

Comment:
If we have this document template - do we need to use the sampling loader? I would think we could just save this 'Document' as an alias and use the 'Loader' throughout? For some of these we want to exactly repeat the same document, for which we could hard-code one example.

Reply:
I considered using a hard-coded example and decided against it because it would prevent easy experimentation with different object widths or field distributions. The Loader could be used for inserting docs with randomized schema but I believe it would be easier to understand and maintain the workload if the same actor was used for the inserts we are trying to measure. The sampling loader, inho, fits quite well the scenarios here.